lib/IgnorePlugin.js

```diff
@@ -58,10 +58,10 @@ class IgnorePlugin {
 
 	apply(compiler) {
 		compiler.hooks.normalModuleFactory.tap("IgnorePlugin", (nmf) => {
-			nmf.plugin("before-resolve", this.checkIgnore);
+			nmf.hooks.beforeResolve.tapAsync("IgnorePlugin", this.checkIgnore);
```

Comment:
Use `tap` instead of `tapAsync` and remove the `callback`. Return the result instead.